src/datastructure/Route.cpp

```diff
@@ -120,6 +121,106 @@ bool cda_rail::Route::check_consistency(const cda_rail::Network& network) const
     return true;
 }
 
+double cda_rail::Route::length(const cda_rail::Network &network) const {
+    /***
+     * Returns the length of the route, i.e., the sum of the lengths of all edges.
+     * Throws an error if an edge does not exist in the network.
+     *
+     * @param network The network to which the route belongs.
+     * @return The length of the route.
+     */
+
+    double length = 0;
+    for (int i = 0; i < edges.size(); i++) {
+        length += network.get_edge(edges[i]).length;
+    }
+    return length;
+}
+
+void cda_rail::Route::update_after_discretization(const std::vector<std::pair<int, std::vector<int>>> &new_edges) {
+    /**
+     * This method updates the route after the discretization of the network accordingly.
+     * For every pair (v, {v_1, ..., v_n}), v is replaced by v_1, ..., v_n.
+     *
+     * @param new_edges The new edges of the network.
+     */
+
+    const auto old_edges = edges;
+    edges.clear();
```

Comment:
I think the copy can be avoided. You could make a new vector and move it into the old one maybe.

Reply:
Done